content/blog/new-graphic-features/index.Rmd

```diff
@@ -0,0 +1,301 @@
+
+## A fragmented future
+
+With the division of responsibility described above, there are many points in the pipeline that may impose limitations in functionality. The graphic system might not provide the higher-level API to use functionality in the graphics engine, or a graphic device might not provide support for instructions coming in from the graphics engine. While this has mainly been a hypothetical situation prior to R 4.1, it is now the new reality. The new features in the graphics engine was implemented along with high-level support in grid, and low level support in the pdf device along with the cairo based devices. This leaves base plot in the dark and also exclude a range of built-in devices, including the default devices on Windows and macOS. At this point, where high level support from e.g. ggplot2 is still not present, it might not be a big problem as you will probably use these features quite deliberately and know their limitations in support. In the future, however, this could lead to surprises.
+
+As users, this fragmentation is most apparent in the choice of graphic device. After all, you don't expect the graphic system to be capable of something outside of its API, simply because new features were announced for the graphics device. However, if a graphic device lack support it will simply not use the new features and you may end up surprised at what it renders.
+
+When it comes to graphic systems you can expect that grid will be the first, perhaps only, system that ends up supporting new features in the graphics engine. Part of the reason for that is that the grid API is more powerful in general and as new and more complex graphic powers are exposed it can be easiest to make them fit into the most expressive API. This is definitely the case for the latest batch of new features, but I also expect it to be the case going forward. Just because a functionality is exposed in grid, doesn't mean that it can easily be handled in e.g. ggplot2. I'll address what the new features may mean for the future of ggplot2 in the end of the post.
+
+For graphic devices the water is more muddled. Not all devices are under active development and those devices are unlikely to add support for new features. Further, it may be that a graphics device writes to a format, or uses a library, that do not support a new feature provided by the graphics engine. The bottom line is that we can expect an increased fragmentation of the graphics devices in R in terms of which will be up to specs with the latest graphics engine features. It appears that the cairo based devices along with the pdf device from grDevices can be expected to stay current. On our end, we will do our best to make sure that our graphic device packages (currently ragg and svglite) will stay on top of any new additions to the graphics engine.
```

Comment:
```suggestion
For graphic _devices_ the water is more muddled. Not all devices are under active development, and such devices are unlikely to add support for new features. Further, it may be that a graphics device writes to a format, or uses a library that does not support a new feature provided by the graphics engine. The bottom line is that we can expect an increased fragmentation of the graphics devices in R in terms of which will be up to spec with the latest graphics engine features. It appears that the cairo-based devices along with the pdf device from grDevices can be expected to stay current. On our end, we will do our best to make sure that our graphic-device packages (currently ragg and svglite) will stay on top of any new additions to the graphics engine.
```

Reply:
Emphasizing devices (since systems and devices are different, but sound so darn similar to the reader). Also adding hyphens for compound modifiers.